BitwardenShared/Core/Vault/Repositories/VaultRepository.swift

```diff
@@ -1042,6 +1048,10 @@ extension DefaultVaultRepository: VaultRepository {
         try await stateService.getDisableAutoTotpCopy()
     }
 
+    func isVaultEmpty() async throws -> Bool {
+        try await cipherService.fetchAllCiphers().isEmpty
+    }
```

Comment:
🤔 I think fetching all ciphers just to know if they are empty may have a big memory overhead and "slow" performance, specially on large vaults. We could just use `countForFetchRequest` on the data store and check the count which is much faster and memory efficient than fetching objects.